database/src/it/scala/za/co/absa/atum/database/runs/WriteCheckpointTest.scala

```diff
@@ -0,0 +1,280 @@
+    """
+      |{
+      |  "version": 1,
+      |  "keys": ["key1", "key3", "key2", "key4"],
+      |  "values": {
+      |    "key1": "valueX",
+      |    "key2": "valueY",
+      |    "key3": "valueZ",
+      |    "key4": "valueA"
+      |  }
+      |}
+      |""".stripMargin
+  )
+
+  test("Write new checkpoint without data") {
+    val uuid = UUID.randomUUID
+    val startTime = OffsetDateTime.parse("1992-08-03T10:00:00Z")
+    val endTime = OffsetDateTime.parse("2022-11-05T08:00:00Z")
+
+
+    table("runs.partitionings").insert(
+      add("partitioning", partitioning)
+        .add("created_by", "John von Neumann")
+    )
+    //DBTable's insert doesn't return the values yet correctly
+    val fkPartitioning: Long = table("runs.partitionings").fieldValue("partitioning", partitioning, "id_partitioning").get.get
+
+    table("runs.measure_definitions").insert(
+      add("fk_partitioning", fkPartitioning)
+        .add("measure_name", "avg")
+        .add("measured_columns", CustomDBType("""{"col1"}""", "TEXT[]"))
+        .add("created_by", "Aristoteles")
+    )
+
+    assert(table("runs.checkpoints").count(add("fk_partitioning", fkPartitioning)) == 0)
+
+    function("runs.write_checkpoint")
+      .setParam("i_partitioning", partitioning)
+      .setParam("i_id_checkpoint", uuid)
+      .setParam("i_checkpoint_name", "Empty path")
+      .setParam("i_process_start_time", startTime)
+      .setParam("i_process_end_time", endTime)
+      .setParam("i_measurements", CustomDBType("{}", "JSONB[]"))
+      .setParam("i_measured_by_atum_agent", true)
+      .setParam("i_by_user", "J. Robert Oppenheimer")
+      .execute { queryResult =>
+        assert(queryResult.hasNext)
+        val row = queryResult.next()
+        assert(row.getInt("status").contains(11))
+        assert(row.getString("status_text").contains("Checkpoint created"))
+      }
+
+    assert(table("runs.measure_definitions").count(add("fk_partitioning", fkPartitioning)) == 1)
+    assert(table("runs.measurements").count(add("fk_checkpoint", uuid)) == 0)
+    assert(table("runs.checkpoints").count(add("fk_partitioning", fkPartitioning)) == 1)
+    table("runs.checkpoints").where(add("fk_partitioning", fkPartitioning)) {resultSet =>
+      val row =resultSet.next()
+      assert(row.getString("checkpoint_name").contains("Empty path"))
+      assert(row.getOffsetDateTime("process_start_time").contains(startTime))
+      assert(row.getOffsetDateTime("process_end_time").contains(endTime))
+      assert(row.getBoolean("measured_by_atum_agent").contains(true))
+      assert(row.getString("created_by").contains("J. Robert Oppenheimer"))
+      assert(row.getOffsetDateTime("created_at").contains(now()))
+    }
+
+  }
+
+  test("Write new checkpoint"){
+    val uuid = UUID.randomUUID
+    val user = "Franz Kafka"
+    val startTime = OffsetDateTime.parse("1992-08-03T10:00:00Z")
+    val endTime = OffsetDateTime.parse("2022-11-05T08:00:00Z")
+    val measurements =
```

Comment:
This will need to be changed here and on a few places, as the SQL function `write_checkpoint` changed: 


https://github.com/AbsaOSS/atum-service/pull/119/files/bf7a285f38696745c08d91eb05844b9bf00f6044..323035c4fc629283b7a809d9865b2574b986c840#diff-e68361afcc3cd98b5764ea471e94ea754eba71c736ca6ef43d5e9cc7a23e8482